src/datasets/summarizations/trend-partial.summarization.ts

```diff
@@ -0,0 +1,103 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  PointMembershipFunction,
+  MembershipFunction,
+  trapmf,
+  trapmfL,
+  trapmfR,
+} from './libs/protoform';
+import {
+  exponentialMovingAverage,
+  normalizedUniformPartiallyLinearEpsApprox,
+  TimeSeriesPartialTrend,
+} from './libs/trend';
+import { formatX, formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {
+  return cacheSummaries(() => {
+
+    const smoothedPoints = exponentialMovingAverage(points);
+    const partialTrends = normalizedUniformPartiallyLinearEpsApprox(smoothedPoints, 0.01);
+
+    const applyTrendAngleWithWeight = (f: MembershipFunction) => ({ cone }: TimeSeriesPartialTrend) => {

```

Comment:
Please add description for this function

Reply:
Sorry, I misnamed this function. I moved it to `libs/trend.ts` and rename it to `mapConeAngle`.